src/main/scala/nodes/images/Convolver.scala

```diff
@@ -101,29 +101,31 @@ object Convolver {
       whitener: Option[ZCAWhitener],
       varConstant: Double): DenseMatrix[Double] = {
     var x,y,chan,pox,poy,py,px = 0
-    while (x < resWidth) {
-      y = 0
-      while (y < resHeight) {
-        chan = 0
-        while (chan < imgChannels) {
-          poy = 0
-          while (poy < convSize) {
-            pox = 0
-            while (pox < convSize) {
+
+    poy = 0
+    while (poy < convSize) {
```

Comment:
If you are okay with a function call overhead and handling return types somehow you can do that with a simple util function which looks like. 
```
specialFor(0 until amax, 0 until bmax, fn: (a: T, b: T) => U) = {
   while (avar < amax) {
     bvar = 0
     while(bvar < bmax) {
       fn(avar, bvar)
       avar = avar + 1 
    }
     avar = avar + 1
  }
}
```